File: Python/program8.py
ERR_INVALID_AGE = -1

def CalculateTicket(age):
    price = 0

    if age < 0:
        return ERR_INVALID_AGE

    if 0 <= age <= 5:
        price = 0
    elif 5 < age <= 18:
        price = 700
    elif 18 < age <= 50:
        price = 999
    elif age > 50:
        price = 500

    return price


def main():
    try:
        age = int(input("Please enter your age : "))
        ticket_price = CalculateTicket(age)

        if ticket_price == ERR_INVALID_AGE:
            print("Unable to calculate the ticket price as age is invalid")
        else:
            print(f"Your ticket price is : {ticket_price}")
    except ValueError:
        print("Invalid input! Please enter a valid number for age.")


if __name__ == "__main__":
    main()

File: JAVA/program8.java
import java.util.*;
class CalculateTicket
{
    public static final int ERR_INVALID_AGE = -1;

    public int TicketPrice( int iAge)
    {
        int iPrice = 0;

        if(iAge < 0)
        {
            return ERR_INVALID_AGE;
        }

        else if((iAge >= 0) && (iAge <= 5))
        {
            iPrice = 0;
        }

        else if((iAge > 5) &&(iAge <= 18))
        {
            iPrice = 700;
        }

        else if((iAge > 18) && (iAge <= 50))
        {
            iPrice = 999;
        }

        else if(iAge > 50)
        {
            iPrice = 500;
        }

        return iPrice;

    }
}

class program8
{
    public static void main(String[] A)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iRet = 0;

        System.out.println("Enter your age :");
        iValue = sobj.nextInt();

        CalculateTicket cobj = new CalculateTicket();
        iRet = cobj.TicketPrice(iValue);

        if(iRet == cobj.ERR_INVALID_AGE)
        {
            System.out.println("Unable to count ticket price because your age is invalid:");
        }
        else
        {
            System.out.println("Your ticket price is : "+ iRet);
        }

    }
}